helpers/metric.go

```diff
@@ -67,5 +67,8 @@ func ReadMetric(m prometheus.Metric) *Metric {
 	if pb.Untyped != nil {
 		return &Metric{name, help, labels, dto.MetricType_UNTYPED, pb.GetUntyped().GetValue()}
 	}
+	if pb.Histogram != nil {
+		return &Metric{name, help, labels, dto.MetricType_HISTOGRAM, pb.GetHistogram().GetSampleSum()}
```

Comment:
Histogram just doesn't fit that simple `Metric` type – comparing histograms by sums only is not enough. If we want to go in that direction, then we need to expand that type.

I think the better way would be to compare representations in Prometheus text exposition format like there:
* https://github.com/percona/rds_exporter/blob/master/enhanced/metrics_test.go
* https://github.com/percona/rds_exporter/blob/master/enhanced/metrics_data_test.go

But please do not work on that now. PMM 2.0 core takes priority now. Exporters should wait.

Reply:
Another option is to use a new Prometheus package for the same result: https://godoc.org/github.com/prometheus/client_golang/prometheus/testutil

Reply:
Oh, we already have a helper for that: https://github.com/percona/exporter_shared/blob/3897bc6f2200ebbb8fba48bdd8842f3fcc10b92d/helpers/format.go#L41

@DexterHD I think we could mark `Metric` type as deprecated.

Reply:
And if someone can send PR replacing `NewRegistry` in `Format` with `NewPedanticRegistry`, that would be great! :)

Reply:
@AlekSi This change was added, not for comparing histogram values, but only to avoid [TestCollector](https://github.com/percona/mongodb_exporter/blob/af193c701473eb4c9c5f854f003ae16d0673e005/collector/mongodb_collector_test.go#L74-L79) from FAIL. All we do in that test is counting a number of metrics and checking mongodb_version_info. 😕 
We can't merge https://github.com/percona/mongodb_exporter/pull/134 at least because **TestCollector** is FAILs without that change. Maybe we can find another way to fix that test?